### DISCUSSION TITLE
How to disable typechain by default?

### DISCUSSION THREAD

**Discussion by AlissonRS** [NomicFoundation/hardhat#4736] [Troubleshooting]
When I run `hardhat compile` task, it automatically generates types via typechain:

```
Generating typings for: 119 artifacts in dir: src/types for target: ethers-v5
Successfully generated 316 typings!
Compiled 109 Solidity files successfully (evm target: paris).
```

I don't want this behavior. If I want types generated, I'd like to explicitly run `hardhat typechain` task for that.

The reason is that I have the types under version control because I distribute them via NPM package. However, we also use `solidity-coverage` to run coverage on our unit tests, and because it uses different compiler configs than we have in our `hardhat.config.ts` file, it modifies types with different bytecode, so devs can accidentally push types with the bytecode generated by coverage task.

From what I know, it's not possible to disable type generation when running `hardhat coverage` as it just runs the `compile` task internally before running.

**Comment by fvictorio** [comment]
Hi @AlissonRS, sorry for not responding sooner!

As far as I know, there's no way in typechain to configure this. But you can disable it yourself by overriding the compile task:

```typescript
task("compile").setAction(async function (args, hre, runSuper) {
  return runSuper({ ...args, noTypechain: true });
});
```

Is that workaround enough for you?
